gen/common/util.mli

```diff
@@ -32,7 +32,38 @@ val arch_opt : Archs.t ref -> spec
 
 val parse_cmdline : spec list -> (string -> unit) -> unit
 
+module type Monad = sig
+  type 'a t
+  val pure : 'a -> 'a t
```

Comment:
I assume `pure` and`return` are the same concept, the latter name is the function name in the ocaml tutorial ? (https://ocaml.org/docs/monads)